source/module_esolver/esolver_ks.cpp

```diff
@@ -560,6 +562,12 @@ namespace ModuleESolver
     {
         int precision = 3;
         std::string tag = "POT";
+// add nkstot��nkstot_ibz to output json
+#ifdef __RAPIDJSON
+        int Jnkstot = this->pelec->klist->nkstot;
+        int Jnkstot_ibz = this->pelec->klist->nkstot_ibz;
+        Json::add_nkstot(Jnkstot,Jnkstot_ibz);
+#endif //__RAPIDJSON 
```

Comment:
Suggest adding those codes at the end of function https://github.com/deepmodeling/abacus-develop/blob/dd227ae685d4b0c7e3f43513f437aa3b08b7bead/source/module_esolver/esolver_ks.cpp#L72

Reply:
It will error due to the klist is not initialized.
![image](https://github.com/deepmodeling/abacus-develop/assets/57103981/87b96231-df99-42b0-9b6f-03218d6574df)

Reply:
Sorry, I do not know this.  The klist should has been initialized at the beginning of function `run`. You can add those codes in this function 
 https://github.com/deepmodeling/abacus-develop/blob/dd227ae685d4b0c7e3f43513f437aa3b08b7bead/source/module_esolver/esolver_ks.cpp#L347